### PR TITLE
implements new echo instruction pipeline

### PR DESCRIPTION
**Pull request by Benedict-Carling**
Addresses #134 

```python
import basicsynbio as bsb
import zipfile
import os
import pandas as pd
from pathlib import Path

utr_linkers = ["UTR1-RBS1"]
promoter_assemblies = []
for utr_linker in utr_linkers:
    promoter_assemblies += [
        bsb.BasicAssembly(
            f"promoter_construct_{ind}_{utr_linker}",
            bsb.BASIC_SEVA_PARTS["v0.1"]["26"],
            bsb.BASIC_BIOLEGIO_LINKERS["v0.1"]["LMP"],
            promoter,
            bsb.BASIC_BIOLEGIO_LINKERS["v0.1"][utr_linker],
            bsb.BASIC_CDS_PARTS["v0.1"]["sfGFP"],
            bsb.BASIC_BIOLEGIO_LINKERS["v0.1"]["LMS"],
        )
        for ind, promoter in enumerate(bsb.BASIC_PROMOTER_PARTS["v0.1"].values())
    ]
my_build = bsb.BasicBuild(*promoter_assemblies)
echozippath = bsb.export_echo_assembly(
    my_build, alternate_well=True, clips_plate_size=1536
)
```
An example of the new zip file that is created when the above code is ran

[Echo_Instructions_17-06-2021_16.54.28.zip](https://github.com/LondonBiofoundry/basicsynbio/files/6671574/Echo_Instructions_17-06-2021_16.54.28.zip)

the following three arguments are added to echo_instructions

-     assemblies_per_clip: int = 28,
-     clips_plate_size: Literal[6, 24, 96, 384, 1536] = 384,
-     assemblies_plate_size: Literal[6, 24, 96, 384, 1536] = 96,

the following arguments are added to csv_export

-     clip_plate_mapping: dict = None,
-     assembly_plate_mapping: dict = None,

The original echo_instruction unit tests have not been changed and still pass which is ideal, and new unit tests are added to check the new columns of the csv with and without alternate_wells

### CODE REVIEW

**Comment by hainesm6** [comment]
Some nice work @Benedict-Carling! Just wondering if it's feasible to do type hinting on the `clip_plate_mapping` and `assembly_plate_mapping` arguments? Just trying to figure out the format of them.

**Comment by Benedict-Carling** [comment]
Hi @hainesm6 Just did that!  the clips contains a list as the same clip can be in multiple wells - the assemblies one is just a str as I don't think the same unique assembly can be in different wells if I'm not mistaken.

File: basicsynbio/cam/echo_instructions.py [hunk 1]
```diff
@@ -30,6 +37,12 @@ def export_echo_assembly(
         buffer_well (optional): location in 6 well plate of assembly buffer.
         water_well (optional): location in 6 well plate of dH20.
         alternate_well (optional): specifies whether alternating wells are to be used in the input 384 well plate.
+        assemblies_per_clip (optional): specifies the number of clip reactions (to create assemblies) one filled
```
Comment (hainesm6) [inline basicsynbio/cam/echo_instructions.py:40]: We can use the same description for the assemblies_per_clip argument as used in the pdf_instructions function.

**Review by hainesm6 — CHANGES_REQUESTED** [review]
Just the one minor comment @Benedict-Carling. Great work!

File: basicsynbio/cam/echo_instructions.py (hunk 1 above)
Comment (Benedict-Carling) [reply basicsynbio/cam/echo_instructions.py]: copy and pasted the one from pdf_instructions over 👍

**Review by hainesm6 — APPROVED** [review]

**Pull request merged**